insalan/tournament/views.py

```diff
@@ -309,8 +319,89 @@ class TeamDetails(generics.RetrieveUpdateDestroyAPIView):
 
     queryset = Team.objects.all().order_by("id")
     serializer_class = serializers.TeamSerializer
-    permission_classes = [permissions.IsAdminUser | ReadOnly]
+    permission_classes = [permissions.IsAdminUser | ReadOnlyPlusPatch]
```

Comment:
Couldn't you define
```py
class Patch(BasePermission):
    """Is the request using HTTP Method PATCH"""

    def has_permission(self, request, _view):
        """Define the permissions for this class"""
        return request.method == "PATCH"
```

and chain them?
```py
     permission_classes = [permissions.IsAdminUser | ReadOnly | Patch]
```

It would make more sense IMO

Reply:
Yep, you're right, will be fixed.